attributes/source.rb

```diff
@@ -21,12 +21,16 @@
 
 include_attribute 'chef_nginx::default'
 
-default['nginx']['init_style'] = if node['platform'] == 'ubuntu' && node['platform_version'].to_f <= 14.04
-                                   # init_package identifies 12.04/14.04 as init, but we should be using upstart here
-                                   'upstart'
-                                 else
-                                   node['init_package']
-                                 end
+if node['platform'] == 'ubuntu'
+  if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
+    default['nginx']['init_style'] = 'systemd'
```

Comment:
I think node['init_style'] should be used here. So rewrite the if statement to atch only between 14.04 and 14.10